src/robot/parsing/model/statements.py

```diff
@@ -236,6 +285,44 @@ def args(self):
 class Documentation(DocumentationOrMetadata):
     type = Token.DOCUMENTATION
 
+    @classmethod
+    def from_params(
+            cls,
+            doc,
+            test_documentation=False,
+            split_on_eol=True,
```

Comment:
I agree on `settings_section` (with default to True, since it's opposite of current flag meaning). 

I will use `split_multiline` since I'm using this name somewhere else too, and move to the end of args.